google/resource_google_organization_policy_test.go

```diff
@@ -0,0 +1,311 @@
+				Check:  testAccCheckGoogleOrganizationListPolicyAll("listAll", "ALLOW"),
+			},
+		},
+	})
+}
+
+func TestAccGoogleOrganizationPolicy_list_allowSome(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+	org := os.Getenv("GOOGLE_ORG")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleOrganizationPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGoogleOrganizationPolicy_listSome(org, "allow"),
+				Check:  testAccCheckGoogleOrganizationListPolicyAllowedValues("listSome", SOME_ORG_POLICIES),
+			},
+		},
+	})
+}
+
+func TestAccGoogleOrganizationPolicy_list_denyAll(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+	org := os.Getenv("GOOGLE_ORG")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleOrganizationPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGoogleOrganizationPolicy_listAll(org, "deny"),
+				Check:  testAccCheckGoogleOrganizationListPolicyAll("listAll", "DENY"),
+			},
+		},
+	})
+}
+
+func TestAccGoogleOrganizationPolicy_list_denySome(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+	org := os.Getenv("GOOGLE_ORG")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleOrganizationPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGoogleOrganizationPolicy_listSome(org, "deny"),
+				Check:  testAccCheckGoogleOrganizationListPolicyDeniedValues("listSome", SOME_ORG_POLICIES),
+			},
+		},
+	})
+}
+
+func TestAccGoogleOrganizationPolicy_list_update(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+	org := os.Getenv("GOOGLE_ORG")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleOrganizationPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGoogleOrganizationPolicy_listAll(org, "allow"),
+				Check:  testAccCheckGoogleOrganizationListPolicyAll("listAll", "ALLOW"),
+			},
+			{
+				Config: testAccGoogleOrganizationPolicy_listSome(org, "deny"),
+				Check:  testAccCheckGoogleOrganizationListPolicyDeniedValues("listSome", SOME_ORG_POLICIES),
+			},
+		},
+	})
+}
+
+func testAccCheckGoogleOrganizationPolicyDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_organization_policy" {
+			continue
+		}
+
+		org := "organizations/" + rs.Primary.Attributes["org_id"]
+		constraint := rs.Primary.Attributes["constraint"]
+		_, err := config.clientResourceManager.Organizations.GetOrgPolicy(org, &cloudresourcemanager.GetOrgPolicyRequest{
+			Constraint: constraint,
+		}).Do()
+
+		if err != nil {
```

Comment:
wait, shouldn't this be the opposite? I'd expect a nil error to mean it exists.

Reply:
Good catch. I fixed it. It basically always returns an org policy. If it has been cleared, both the boolean and list policy field are `nil`.